Update plex.py

raddatzk opened the pull request:
show information about media depending if it is a movie, an episode, audio or picture
set time_between_scans to 10 s to match with plex media_player component

## Description:
![plex_activity_monitor](https://user-images.githubusercontent.com/8616693/36974355-ad944a5c-2076-11e8-8800-fc7458551356.PNG)

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#<home-assistant.github.io PR number goes here>

## Example entry for `configuration.yaml` (if applicable):
```yaml

```

## Checklist:
  - [x] The code change is tested and works locally.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

raddatzk committed:
Update plex.py

show information about media depending if it is a movie or an episode
set time_between_scans to 10 s to match with plex media_player component

homeassistant commented:
Hi @chronm, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#12157).

Once you do that we will be able to review and accept this pull request.

Thanks!

raddatzk committed:
Update plex.py

lint

raddatzk committed:
Update plex.py

linting

raddatzk committed:
Update plex.py

linting

raddatzk committed:
Update plex.py

linting

ryanm101 reviewed (approved):
Overall looks good

danielperna84 commented on homeassistant/components/sensor/plex.py:
```diff
@@ -116,8 +116,25 @@ def update(self):
-            title = sess.title if sess.title is not None else ""
-            year = sess.year if sess.year is not None else ""
-            now_playing.append((user, "{0} ({1})".format(title, year)))
+            if sess.TYPE == 'movie':
```
What happens if neither a movie nor an episode is played? I use Plex as the source for music. Haven't looked deeper into how Plex works, but I think there should be an `else` for everything besides movies and episodes.

chronm replied:
Oh you are totally right. I will work on it the next days

ryanm101 replied:
good catch.. and this is why i should not review code straight after work :)

raddatzk committed:
Update plex.py

added catch  for tracks and everything else
if no release year is given, instead of () it now show nothing

trisk commented:
Minor formatting nits: With episode titles or other info potentially empty, it might make sense to append to the title string rather than use a single `format()` call.

Also, if you want to be more consistent with display names in the Plex UI, it looks like audio track titles don't include the album release year, and you could use something like `"S%d • E%d" % (sess.parentIndex, sess.index)` instead of `sess.seasonEpisode`.

chronm commented:
Good idea with appending and seasonEpisode, but I am not sure if I get what you mean with the audio tracks. The release year ist displayed in the second section next to the album title

trisk commented:
@chronm This may have changed in recent versions of Plex since I haven't tried importing music in a while, but I recall the "Now Playing" info used to have just the artist, album, and title without the year.

chronm commented:
Oh I think I misunderstood you. I thought you meant the album year is missing in the plugin, but you mean the Plex UI, right? 
So better remove this information?

trisk commented:
@chronm Since this is for human-readable purposes, my opinion is that it looks nicer to be consistent with what Plex displays.

raddatzk committed:
Update plex.py

Remove the album year to match with the Plex UI

chronm commented:
Unfortunately, for example when I have two streams running from the same user, it just shows one of them.
I think this comes from the returned dictionary with the username and corresponding stream, where it gets overwritten.
Do you have an idea to solve this?

trisk commented:
@chronm Instead of using the username as the key for the dict returned from `device_state_attributes`, perhaps the device instead?

chronm commented:
How about a combination of username and device?
![unbenannt](https://user-images.githubusercontent.com/8616693/36285281-4134c3ec-12ab-11e8-9378-1a8caa3d8cf4.PNG)

trisk commented:
That sounds good. Plex's "Now Playing" page (https://app.plex.tv/desktop#!/status/playing) lists streams by player and user. Also I just noticed that Plex has a separator between episode numbers and titles in the "Continue Watching" list, but not in "Now Playing". I think either way is fine.

I don't know what is causing the text alignment to change in the card, but it's probably not something the HA backend can do anything about.

chronm commented:
I don't understand, why Travis py34 is not working. Could you have a look at the log?

raddatzk committed:
Update README.rst

raddatzk committed:
Update README.rst

chronm commented:
So I added a line in README.rst , let it run a test and now removed it, and now it passes the test. I don't really understand where this comes from

trisk commented:
Test was failing randomly because the VM took too long, that can be ignored.

chronm commented:
Okay, so if one test fails, first restart it?

trisk commented:
You should check the Travis log for the failure first, but often if all environments except one pass, it's a problem with the test environment rather than the code.

chronm commented:
Thanks a lot for the information

balloob commented on homeassistant/components/sensor/plex.py:
```diff
@@ -102,7 +102,9 @@ def state(self):
     def unit_of_measurement(self):
         """Return the unit this state is expressed in."""
-        return "Watching"
+        if self._state > 1:
```
Not allowed. unit needs to be static.

chronm replied:
Okay, I will change it, but what is the reason?

balloob commented on homeassistant/components/sensor/plex.py:
```diff
@@ -116,8 +118,66 @@ def update(self):
+                # example
+                # "The Incredible Hulk (2008)"
+                mov_title = (sess.title
+                             if sess.title is not None
```
These inline if's make for very unreadable code, please rewrite it.

balloob commented on homeassistant/components/sensor/plex.py:
```diff
@@ -116,8 +118,66 @@ def update(self):
+                                if sess.grandparentTitle is not None
+                                else None)
+
+                track_album = (sess.parentTitle
```
There is no reason for this if…else. If it's `None`, you assign `None`. So you can just assign sess.parentTitle ?

chronm commented:
Can I assume, that 
- a tv-show / episodes
- a movie
- an artist / album / track
alwas has a name?

ryanm101 commented:
@chronm it's a safe assumption until we get an issue saying it's not :)

raddatzk committed:
Update plex.py

reformat code to make it more readable
recorded tv shows might not have episode numbers assigned -> check before adding to title

ryanm101 reviewed (changes requested):
Please cleanup excessive whitespace


e.g. Lines 143-148

raddatzk committed:
Update plex.py

cleanup excessive whitespace

raddatzk committed:
Update plex.py

cleanup excessive whitespace
